Feature/frontend

Viiiin94 opened the pull request:
# Summary
로그인 토큰 관련해서 PR

# Relevant

# Screenshot

체크 방법은 [ ] 괄호 안에 x를 기입하시면 됩니다.

- [] 작업한 내용에 대해 테스트를 진행했음.
- [x] 칸반보드에 해당 작업이 업데이트 되어있음.
- [] commit 전, `git status`, `git pull`을 통해 반영되지 않은 코드가 없음을 확인함.

# To Reviewers
리뷰어들이 집중해서 봐주었으면 하는 사항이 있다면 적어주세요.

JmKanmo commented:
application.yml 파일에 datasource 관련 설정은 저희 작업환경이 되도록 통일되어야하는데 
제가 임의로 변경했는데 아래와 같이 공통 작업환경 db인 h2 inmemory db에서 수행되도록 수정하겠습니다. 필요에따라 각자 환경에 맞게 실행하면  될것같습니다. 제가 다시 커밋하고 develop에 머지하도록 할게요. 


  h2:
    console:
      enabled: true
      path: /h2-console

  datasource:
    url: jdbc:h2:mem:testdb
    driver-class-name: org.h2.Driver
    hikari:
      username: sa
      password:

Haze-S commented on src/main/java/com/service/zerobnb/web/room/domain/LodgmentTypeRoom.java:
```diff
@@ -0,0 +1,30 @@
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class LodgmentTypeRoom extends BaseTimeEntity {
```
내일 회의에서 accommodation, lodgementType, rentType 에 대해 자세히 알려주실 수 있을까요? 
저는 각각의 장소들이 다 같은 숙박업소여서 테이블을 나눠서 만들 필요가 있을까? 란 생각이 들어서 준모님 생각을 듣고싶어요! 
숙소 테이블 하나에 type 컬럼을 하나 넣으면 될 것 같아서요~ 다른 분들의 의견도 같이 들어봐요 :)

JmKanmo replied:
야놀자의 경우를 참고해보니,    아래와 같이 2가지로 나뉘더군요. 
다른 어플 에어비엔비의 경우는 그렇지 않았고. 
야놀자의 경우는 호텔,모텔 등이 잇어서 대실이 추가된거고 
그외 나머지는 거의 숙박(몇일동안 머무를지)으로 나뉘는 것 같습니다. 
그래서  대실,숙박 2가지로 나뉘었습니다. 
대실, 숙박

Haze-S reviewed (commented):
PopularFacilityServiceType과 ReviewType 에 대한 설명도 같이 듣고싶어요 😄 
각 테이블에 타입을 하나씩 저장한 후에 불러오는 건가요?
그럼 한 컬럼에 타입들을 다 넣는건가요?
리뷰 타입은 리뷰 타입별로 점수를 주는 걸까요?

코드를 보니 질문이 많아지네요 ㅎㅎ 
확실히 알고 가야 다음 스텝이 명확해질 것 같아서 여쭤봅니다!
기분 나빠하지 않으셨으면 좋겠어용 🙏

MyungHwanKim commented:
코드 작성하시느라 고생하셨어요👍👍
전반적인 코드 잘 확인했습니다!!

궁금한 사항 몇 가지가 작성해볼게요.

MyungHwanKim commented on src/main/java/com/service/zerobnb/util/status/CouponType.java:
```diff
@@ -0,0 +1,24 @@
+package com.service.zerobnb.util.status;
+
+
+public enum CouponType {
+    /**
+     * 대실
+     */
+    RENT("rent"),
+
+    /**
+     * 숙박
+     */
+    LODGMENT("lodgment"),
+
+    /**
+     * 제한 없음
+     */
+    ALL("all");
+    private final String type;
+
+    CouponType(String type) {
+        this.type = type;
+    }
+}
```
CouponType에서 RENT가 있는데 RENT는 AccommodationType에서 MOTEL에만 적용되지 않을까 생각합니다. 저희가 진행하는 프로젝트가 숙박업소이지만, 약간 airbnb에 더 가깝다고 AccommodationType에서 MOTEL는 빼도 되지 않을까 생각했습니다.
그러면 CouponType의 class가 필요 없어지고 오로지 일자 별 count로만 나눠주게 될 거 같아 제 생각에는 이 방법도 괜찮다고 생각했습니다.
준모님의 생각은 어떤 지 궁금합니다.

JmKanmo replied:
아무래도 좀 더 간단한게 가는게 좋을 거 같아요.  참고하도록 할게요. 제 생각대로 진행한거라 다른분들 의견이 궁금하긴 했습니다.

MyungHwanKim commented on src/main/java/com/service/zerobnb/web/room/domain/LodgmentTypeRoom.java:
```diff
@@ -0,0 +1,30 @@
+package com.service.zerobnb.web.room.domain;
+
+import com.service.zerobnb.util.BaseTimeEntity;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+
+@Entity
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class LodgmentTypeRoom extends BaseTimeEntity {
```
위에서 말씀드린 CouponType 얘기에 이어 만약 나누지 않게 된다면 LodgmentTypeRoom과 RentTypeRoom도 나눌 필요 없다고 생각합니다!! RENT(대실) 하나로만 하기에는 비중과 코드 상 중복이기도 하나의 Entity로만 구성하는 것이 좋다고 생각합니다. 
그래서 Room Entity에 위 2가지 Type에 관한 컬럼은 없어도 될 것 같다고 생각합니다.

---

이상입니다~!! 다른 분들의 의견도 같이 들어보면서 얘기 나누어봐요😄😄

JmKanmo commented on src/main/java/com/service/zerobnb/util/status/CouponType.java:
```diff
@@ -0,0 +1,24 @@
+package com.service.zerobnb.util.status;
+
+
+public enum CouponType {
```
아 해당 쿠폰의 기능은 대실(할인), 숙박(할인), 제한없음(모두 할인) 의 기능에 맞춰서 제가 작성한건데, 다른분들 의견은 어떤지 궁금하네요. 의견 제공 감사합니다

Viiiin94 committed:
Token Change / NotHeaderRoute Add

회원가입때 생성되는 불필요한 토큰 제거
로그인시 토큰 생성 / 로컬스토리지에 저장
새롭게 라우터를 추가하여 로컬에 저장된 토큰에 따라 페이지 변화

Viiiin94 committed:
Login Token

Login Token & Room list markup
